src/builder/QuarkBuilder.sol

```diff
@@ -261,28 +262,53 @@ contract QuarkBuilder {
     }
 
     // Assert that each chain has sufficient funds to cover the max cost for that chain.
-    // NOTE: This check assumes we will not be bridging payment tokens for the user.
-    function assertPaymentAffordable(
-        TransferIntent memory transferIntent,
-        Accounts.ChainAccounts[] memory chainAccountsList,
-        Payment memory payment
-    ) internal pure {
-        if (payment.isToken) {
-            for (uint256 i = 0; i < payment.maxCosts.length; ++i) {
-                uint256 paymentAssetBalanceOnChain = Accounts.sumBalances(
-                    Accounts.findAssetPositions(payment.currency, payment.maxCosts[i].chainId, chainAccountsList)
-                );
-                uint256 paymentAssetNeeded = payment.maxCosts[i].amount;
+    // Check user account can cover the cost of each actions
+    function assertActionsAffordable(Actions.Action[] memory actions, Accounts.ChainAccounts[] memory chainAccountsList)
+        internal
+        pure
+    {
+        uint256 plannedBridgeAmount = 0;
+        for (uint256 i = 0; i < actions.length; ++i) {
+            address tokenUsed;
+            uint256 amountUsed;
+            if (Strings.stringEqIgnoreCase(actions[i].actionType, Actions.ACTION_TYPE_TRANSFER)) {
+                Actions.TransferActionContext memory transferActionContext =
+                    abi.decode(actions[i].actionContext, (Actions.TransferActionContext));
+                tokenUsed = transferActionContext.token;
+                amountUsed = transferActionContext.amount;
+            } else if (Strings.stringEqIgnoreCase(actions[i].actionType, Actions.ACTION_TYPE_BRIDGE)) {
+                Actions.BridgeActionContext memory bridgeActionContext =
+                    abi.decode(actions[i].actionContext, (Actions.BridgeActionContext));
+                tokenUsed = bridgeActionContext.token;
+                amountUsed = bridgeActionContext.amount;
+                plannedBridgeAmount += amountUsed;
+            } else {
+                revert InvalidActionType();
+            }
```

Comment:
In order to proper know the affordability of the actions when involve bridging + transferring actions on payment token. Will need to add `plannedBridgeAmount` when checking if target chain has enough fund.  
So that when check if target chain has enough fund to cover 5USDC won't fail when target chain only has 3USDC and 2USDC incoming from Base.